tests/simple/testdata/dynamic.textproto

```diff
@@ -1,2 +1,2161 @@
+  }
+  test {
+    name: "field_read_proto3_zero"
+    container: "google.api.expr.test.v1.proto3"
+    expr: "TestAllTypes{single_int32_wrapper: 0}.single_int32_wrapper"
+    value { int64_value: 0 }
+  }
+  test {
+    name: "field_read_proto3_unset"
+    container: "google.api.expr.test.v1.proto3"
+    expr: "TestAllTypes{}.single_int32_wrapper"
+    value { null_value: NULL_VALUE }
+  }
+}
+section {
+  name: "int64"
+  description: "Tests for int64 conversion."
+  test {
+    name: "literal"
+    expr: "google.protobuf.Int64Value{value: -123}"
+    value { int64_value: -123 }
+  }
+  test {
+    name: "literal_no_field_access"
+    expr: "google.protobuf.Int64Value{value: -123}.value"
+    disable_check: true
+    eval_error {
+      errors { message: "no_matching_overload" }
+    }
+  }
+  test {
+    name: "literal_zero"
+    expr: "google.protobuf.Int64Value{}"
+    value { int64_value: 0 }
+  }
+  test {
+    name: "var"
+    expr: "x"
+    type_env {
+      name: "x"
+      ident { type { message_type: "google.protobuf.Int64Value" } }
+    }
+    bindings {
+      key: "x"
+      value { value { object_value {
+            [type.googleapis.com/google.protobuf.Int64Value]
+            { value: 2000000 }
+          } } }
+    }
+    value { int64_value: 2000000 }
+  }
+  test {
+    name: "field_assign_proto2"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_int64_wrapper: 432}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_int64_wrapper { value: 432 } }
+      }
+    }
+  }
+  test {
+    name: "field_assign_proto2_zero"
+    container: "google.api.expr.test.v1.proto2"
+    expr: "TestAllTypes{single_int64_wrapper: 0}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto2.TestAllTypes]
+        { single_int64_wrapper { } }
+      }
+    }
+  }
+  test {
+    name: "field_assign_proto3"
+    container: "google.api.expr.test.v1.proto3"
+    expr: "TestAllTypes{single_int64_wrapper: -975}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto3.TestAllTypes]
+        { single_int64_wrapper { value: -975 } }
+      }
+    }
+  }
+  test {
+    name: "field_assign_proto3_zero"
+    container: "google.api.expr.test.v1.proto3"
+    expr: "TestAllTypes{single_int64_wrapper: 0}"
+    value {
+      object_value {
+        [type.googleapis.com/google.api.expr.test.v1.proto3.TestAllTypes]
+        { single_int64_wrapper { } }
+      }
+    }
+  }
+}
+section {
+  name: "uint32"
+  description: "Tests for uint32 conversion."
+  test {
+    name: "literal"
+    expr: "google.protobuf.UInt32Value{value: 123u}"
+    value { uint64_value: 123 }
+  }
+  test {
+    name: "literal_no_field_access"
+    expr: "google.protobuf.UInt32Value{value: 123u}.value"
+    disable_check: true
+    eval_error {
+      errors { message: "no_matching_overload" }
```

Comment:
Interesting that the `.value` returns no matching overload. I would have expected 'no such field' or 'no such attribute' for tests like this.

Reply:
I would have expected `no_such_field` too, but since I just rewrote the language spec section on field access, I remembered that it first evaluates the expression to the left of the "." then throws `no_matching_overload` if it's not a map or a message.  I don't think we're checking the exact error message yet in the test validation.

Reply:
It's likely that one or both of the following snippets from `cel-go/interpreter/attributes.go` needs to be switched to `no such field`:
- [Resolve](https://github.com/google/cel-go/blob/3023710f92aa1469e7c0734514b080e15fbcb6a5/interpreter/attributes.go#L352)
- [refResolve](https://github.com/google/cel-go/blob/3023710f92aa1469e7c0734514b080e15fbcb6a5/interpreter/attributes.go#L892)

Seems worth a minor PR to fix, maybe. It could be the case that we treat '.' select on simple types as `no such overload`, but I think `no such field` makes more intuitive sense.

Reply:
I'm actually happy with the current error string - the message does have a `value` field after all. Also, it would be strange to treat the given expr any different than `123u.value`. If "select()" were an ordinary function, with overloads `(message, string) -> X` and `(map, string) ->X`, we'd expect `no_matching_overload`.

Having made a case as best I can, I'll leave it to you to make the call.